src/Tools/dotnet-monitor/CollectionRules/Triggers/ICollectionRuleTrigger.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers
+{
+    /// <summary>
+    /// Interface implemented by collection rule triggers.
+    /// </summary>
+    internal interface ICollectionRuleTrigger
+    {
+        /// <summary>
+        /// Waits for the trigger condition to be satisfied.
+        /// </summary>
+        Task WaitForSatisfiedConditionAsync(CancellationToken cancellationToken);
```

Comment:
Given this interface, a trigger can never be satisfied more than once correct? We would need to recreate the trigger after taking the actions?

Reply:
That's not necessarily correct. When called, a TaskCompletionSource could be registered for each call to the method. When the next satisfaction of the trigger condition occurs, then all TCS instances are completed.

I haven't really settled on this being the correct way to do it; we could just have a similar implementation as the pipelines where it is initialized with a callback and then there are start/run/stop/etc methods.

In any event, this interface is likely to change as we go to implement them. I think what's really going to drive the change is the need to say "take this callback and set yourself up" separately from "now it's okay to notify of condition satisfaction".

Reply:
Sounds good. I do think that the callback approach will be much easier to manage for the caller.